### ISSUE TITLE
How does it know which peer to lookup?

### ISSUE THREAD

**Issue by abitrolly** [manuels/wireguard-p2p#12]
If there are multiple people exchanging IPs, how does this script know which peer is correct?

**Comment by manuels** [comment]
The messages are encrypted and signed. And each message has a timestamp.

**Comment by abitrolly** [comment]
Yes, but when two peers are independent and there was no prior key exchange between them, how they select each other among hundreds and thousands of potentially submitted entries?

**Comment by manuels** [comment]
I‘m afraid this is not supported. You need an initial key exchange.
Am 12. Mai 2022, 09:39 +0200 schrieb Anatoli Babenia ***@***.***>:

**Comment by abitrolly** [comment]
Thanks. That's the info that is missing from `README.md`. Specifically, how that initial key exchange takes place, where the key is stored and read from.
